# Material and Methods

## 2.1 Materials 
### 2.1.1 Data sets 

For this project we were provided with four data sets. The first one being the "tcga_tumor_log2TPM" a data frame containing about 60000 rows, representing
the gene expression data from RNA-seq for almost 10,000 TCGA  (The Cancer Genome Atlas) cancer patients, representing 33 different tumor types in the columns in the data frame.

The second data set "tcga_tumor_annotation" is also a data frame which contains all of the patients from the previous datasets with 37 clinical annotations (for example gender, race, tissue type etc.)

The third provided data set is a R object "tcga_tumor_normal_datascience_proj_2022" containing expression data of matched tumor and normal tissue for five tumor types (BRCA, KIRC, LUAD, PRAD, THCA), each of them are specific for one tumour type, each of them have three smaller dataframe showing gene expression of tumor samples and normal samples from the same patients, and clinical annotations for these patients.

Last given data set is a list "hallmarks_genesets" containing 46 Hallmark genesets, each of them contains various amounts of genes associated with this set

### 2.1.2 Used Packages
show a table! 
- where from 
- what they do 
- what we used them for 

## 2.2 Methods 

### 2.2.1 Data Exploration 

 2.2.1.1 Cleaning & Filtering 
 
 - checked for  NAs (why & how) + code chunk + function 
 - biotype filtering (why + how) + function + code chunk 
 - Distribution of gene expression values
 - variance filtering 

 
 2.2.1.2 Discriptive analysis
 
- what is this + what is is used for 
- what we did (violin, boxplot, barplot, histos, variance mean, PCA ... What else ????) + plots, Q: do we here need code chunks ?

   Results: what did we acchive with this, what does it say, ow can we use thius inforamtion + in dicussion if needed point out what is interersting and if this needs further research 
   
2.2.2 Metabolic pathways 
  
- how did we find and retrieved the pathways
- where did we extract them form + how  code chunk ?
- for what + what did we d further with them 
 
2.2.3 Gene set scoring - GSEA, GSVA
 
- what is this 
- what used for generally 
- how did we use it + code chunk + what did we get (probs in results, right?) + plotsss + volcano ? 
- quality control ???

######where does venn diagrmm gooooo

### pan cancer 
### focused analysis 
### GSEA vs GSVA
### correlation analysis 
### regression analysis 


nowww resultsss and info impuut, go sleepyy sleep 
 
