---
title: "VolcanoPlot_backup"
author: "linh"
date: '2022-06-15'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
---
title: "VolcanoPlot"
author: "linh"
date: '2022-06-12'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r compute foldchange}

#log2(TPM)(Transcripts Per Million) refers to how much RNA is present in a sample.
#Log2Foldchange describes how one sample is different from another.

# compute mean gene expression for each condition (normal and tumor)

normal_mean_vector = apply(SkinnyHappyTT, 1, mean)
cancer_mean_vector = apply(SkinnySadTT, 1, mean)


foldchange = cancer_mean_vector - normal_mean_vector

# compute foldchange  log2(FC) = log2(TPMcondition1) - log2(TPMcondition2) = log2(TPMcondition1/TPMcondition2)

#e_normal_mean_vector = 2^normal_mean_vector
#e_cancer_mean_vector = 2^cancer_mean_vector

#foldchange =  e_cancer_mean_vector/e_normal_mean_vector

log2foldchange = log2(foldchange)

foldchange = cancer_mean_vector - normal_mean_vector
log2foldchange = log2(foldchange)

#View(foldchange)

```


```{r compute p-values for each genes}

#When identifying differentially expressed (DE) genes from high-throughput gene expression measurements, we would like to take both statistical significance (such as P-value) and biological relevance (such as fold change) into consideration 

# here we calculate p-values for each genes by wilcoxon test
 
 p_value = c()

 for (i in 1:nrow(SkinnyHappyTT)){
  p_value[i] = wilcox.test(unlist(SkinnyHappyTT[i,]),unlist(SkinnySadTT[i,]),paired = TRUE )$p.value
 }




#View (p_value) 

#correcting p value by Benferroni 

length (SkinnyHappyTT[1,])

corrected_pvalue = p_value*length(SkinnyHappyTT[1,])
#View (corrected_pvalue)




```



```{r plot volcano}


df_volcanoplot = data.frame (foldchange, p_value)

df_volcanoplot$diffexpressed = "NO"

df_volcanoplot$diffexpressed[df_volcanoplot$foldchange > 0.6  & df_volcanoplot$p_value < 0.05]= "UP"

df_volcanoplot$diffexpressed[df_volcanoplot$foldchange < 0.6 & df_volcanoplot$p_value < 0.05]= "DOWN"

View (df_volcanoplot)

#View (df_volcanoplot)

#library (ggplot2)
ggplot (data = df_volcanoplot, aes (x=foldchange, y=-log10(p_value), col = diffexpressed)) + geom_point() + theme_minimal() + scale_color_manual(values = c ("blue","black", "red"))+ theme (text=element_text(size = 8))

CompleatePathwayDF_scale = -log10(CompleatePathwayDF_df)


```




